cmd/frontend/graphqlbackend/site_flags.go

```
@@ -18,7 +18,7 @@ func (r *siteResolver) NeedsRepositoryConfiguration(ctx context.Context) (bool,
 	// 🚨 SECURITY: The site alerts may contain sensitive data, so only site
 	// admins may view them.
 	if err := backend.CheckCurrentUserIsSiteAdmin(ctx); err != nil {
-		return false, err
+		return false, nil
```

Why has this changed?

---

This specifically is needed to prevent this query here 
https://github.com/sourcegraph/sourcegraph/blob/f4f545815853d024a01e0e0f019ff4e47303f2de/web/src/site/backend.tsx#L24 from failing for users.

---

Swallow the backend error seems not ideal here.

Instead the frontend should use `include` in the GraphQL query to request certain fields only when the viewer is an site admin, for example: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@87e08bd8cf8d54a27700dfef313db90959158bd6/-/blob/web/src/user/area/UserArea.tsx#L55:38

---

That sounds like a better approach. Ultimately, users really only need to see the `UpdateCheck` field https://github.com/sourcegraph/sourcegraph/blob/d697c7f74a9621b31e90bd08621df11c769ded2d/cmd/frontend/graphqlbackend/schema.graphql#L3283 

On an unrelated note, it looks like if this query is failing, then other site-wide alerts might not work such as `freeUsersExceeded` which should be shown to all users.

---

Hmm, if only one field is needed, construct another method in TypeScript to request that single field seems cleaner approach? (Either way, we need some form of `if siteAdmin {} else` logic in the TypeScript). Up to you :)

---

I suppose that depends on if `site_alerts` are expected to be scoped to admins only.

---

> I suppose that depends on if `site_alerts` are expected to be scoped to admins only.

That is a good question... 

Have you thought about adding logic between these two lines? https://sourcegraph.com/github.com/sourcegraph/sourcegraph@master/-/blob/cmd/frontend/graphqlbackend/site_alerts.go#L75-79

Right now the entire function returns nothing if the viewer isn't a site admin, but we can proposed some safe-for-everyone-to-see alerts that non-site admins can see (before checking if the viewer is an admin). Not sure this approach would actually work but in theory it should.

---

That was my first approach but I wanted to leverage the existing `site_update_check` code. Adding logic there will also require swallowing this error or rewriting some the Typescript query to avoid requesting data that is only viewable by an admin.

---

Correction: I mean to say if `SiteFlags` are meant to be seen by users. 
https://github.com/sourcegraph/sourcegraph/blob/5e425cfec07790e8f4836738629c5ec2b0dd2e6a/web/src/site/backend.tsx#L24
It seems like this query has a mix of data that should be viewable by a user and by an admin

---

As I understand this has been worked around for now, let's continue discussion in https://github.com/sourcegraph/sourcegraph/issues/10889